Src/TestSample.Grpc.NarrowInt.Tests/Grpc/GrpcInterceptorTests.cs

```diff
@@ -0,0 +1,124 @@
+using FluentAssertions;
+using Grpc.Core;
+using Moq;
+using TestSample.Domain;
+using TestSample.Domain.Exceptions;
+using TestSample.Domain.Users;
+using TestSample.Grpc.proto;
+using TestSample.Tests.Framework.TestServer;
+using Xunit;
+
+namespace TestSample.Grpc.NarrowInt.Tests.Grpc;
+
+public class GrpcInterceptorTests
+{
+    private readonly Mock<IUserService> _mockUserService;
+    private readonly Users.UsersClient _usersClient;
+
+    public GrpcInterceptorTests()
+    {
+        _mockUserService = new Mock<IUserService>();

```

Comment:
I briefly discussed with Dan regarding the concerns I have and here they are with respect to mocking the Domain responses:
Assuming we mock service responses to optimize the test time, in the future it leaves a lot of room for human error when (a) More teams are introduced to this strategy, all scenarios for mock may not be covered when different people write the tests - this will leave potential gaps between actual domain responses and mocked responses,
(b) More logic is added to the domain, failure to add them to the mock will leave a gap
 Since our broad integration tests are not intensive and this is the only place adaptors gets tested, my confidence is low with mocked services. 
We have to ramp up on our Manual testing until the process is proven should we choose to keep it this way - A whole other discussion on how to do manual tests on the pipeline.